CONTRIBUTING.md

```diff
@@ -1,27 +1,92 @@
 # zkEVM contribution guidelines
 
-Thank you for investing your time in contributing to our project! 
+Thank you for investing your time in contributing to our project!
 
-Before starting, we encourage you to take a look to the following documentation as it might help you in your first contribution adventure:
-- https://github.com/privacy-scaling-explorations/zkevm-docs
-- https://github.com/privacy-scaling-explorations/zkevm-specs
+# How you can Contribute
+
+First take a look at our [specs](https://github.com/privacy-scaling-explorations/zkevm-specs), [auditing videos](https://www.youtube.com/watch?v=HhHTho2QZa4), and [zkevm docs](https://github.com/privacy-scaling-explorations/zkevm-docs). It’s very important that you understand at a high-level how the ZK-EVM architecture and all of its components work.
+
+## Best Coding Practices
+
+### Creating PRs
+
+1. Fork the zkevm or specs repos.
+2. Write a simple markdown spec doc about the circuit PR and send to the `zkevm-specs` repo.
+3. For PRs that change significantly the codebase or the way on which some procedure is done (test-suite, cross-crate structures, module refactors) please [file an issue](https://github.com/privacy-scaling-explorations/zkevm-circuits/issues/new/choose) and wait until this feature/proposal gets approved by the repo maintainers.
```

Comment:
I would rephrase this a bit: We encourage everyone to open an issue before getting to work on a PR.  This will be a must for "PRs that change significantly the codebase or the way on which some procedure ..."